src/pybind/mgr/dashboard/frontend/src/app/shared/directives/permission-scope.directive.ts

```diff
@@ -0,0 +1,29 @@
+import { Directive, Input, ElementRef } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+
+@Directive({
+  selector: 'input:not([cdNoPermissionScope]), select:not([cdNoPermissionScope])'
+})
+export class PermissionScopeDirective{
+  private permissions: Permissions;
+
+  @Input() flagDisabled: boolean;
+  @Input() set scope(service: any) {
+    const service_name: keyof Permissions = service
+    if (!this.permissions[service_name].update){
+      this.elementRef.nativeElement.disabled= true
+    }
+    if (this.flagDisabled){
+      this.elementRef.nativeElement.disabled = this.flagDisabled
+    }
```

Comment:
Hi you can shorten this part.

```suggestion
    if (this.flagDisabled || !this.permissions[service_name].update){
      this.elementRef.nativeElement.disabled= true
    }
```

I'm still not sure about the name as `PermissionScopeDirective` looks to me like it would handle different permissions, but it should only be used for fields that can only be used with the update permission or you can overwrite to make the field read only.

How about adding allowed states?

This way it would become something like:
```suggestion
    const perms = this.permissions[service_name]
    const disable = allowed.length === 0 || !allowed.map(state => perms[state]).some(state=>state);
    if (disable){
      this.elementRef.nativeElement.disabled= true
    }
```

Reply:
Sure. I can try that.

Reply:
@Devp00l Hey, I went with a different approach, so can u please take a look again?